go-controller/pkg/ovn/libovsdbops/router.go

```diff
@@ -0,0 +1,107 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+// findRouter looks up the router in the cache and sets the UUID
+func findRouter(nbClient libovsdbclient.Client, router *nbdb.LogicalRouter) error {
+	if router.UUID != "" && !IsNamedUUID(router.UUID) {
+		return nil
+	}
+
+	routers := []nbdb.LogicalRouter{}
+	err := nbClient.WhereCache(func(item *nbdb.LogicalRouter) bool {
+		return item.Name == router.Name
+	}).List(&routers)
+	if err != nil {
+		return fmt.Errorf("can't find router %+v: %v", *router, err)
+	}
+
+	if len(routers) > 1 {
+		return fmt.Errorf("unexpectedly found multiple load balancers: %+v", routers)
+	}
+
+	if len(routers) == 0 {
+		return libovsdbclient.ErrNotFound
+	}
+
+	router.UUID = routers[0].UUID
```

Comment:
I have use cases where I need more than the uuid... how about:
```
router = &routers[0]  
```

Reply:
I will let you change this on your rebase as we discussed.